.github/workflows/build-test-radosgw.yml

```diff
@@ -55,23 +54,55 @@ jobs:
           sudo apt-get update
           sudo apt-get install -y s3cmd
 
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.8
+
+      - name: Run Integration tests
+        run: |
+          mkdir -p integration/storage
+          CONTAINER=$(docker run -d \
+            -v $GITHUB_WORKSPACE/integration/storage:/data \
+            -v $GITHUB_WORKSPACE/ceph/build/bin:/radosgw/bin \
+            -v $GITHUB_WORKSPACE/ceph/build/lib:/radosgw/lib \
+            -p 7480:7480 \
+            quay.io/s3gw/run-radosgw \
+              --rgw-backend-store sfs \
+              --debug-rgw 1)
+
+          sleep 5
+          # ^ This is needed to give the gateway some time to boot up.
+          # It also serves to make sure the gateway takes no more than 5 seconds
+          # to boot up.
+
+          echo "Running Integration Tests:"
+          cd ceph/qa/rgw/store/sfs/tests
+
+          python3 -m pip install -r ./requirements.txt
+          python3 -m unittest test-*.py
+
+          docker kill $CONTAINER
+
       - name: Run smoke tests
         run: |
-          mkdir storage
-          docker run -d \
-            -v $GITHUB_WORKSPACE/storage:/data \
+          mkdir -p smoke/storage
+          CONTAINER=$(docker run -d \
+            -v $GITHUB_WORKSPACE/smoke/storage:/data \
             -v $GITHUB_WORKSPACE/ceph/build/bin:/radosgw/bin \
             -v $GITHUB_WORKSPACE/ceph/build/lib:/radosgw/lib \
             -p 7480:7480 \
             quay.io/s3gw/run-radosgw \
               --rgw-backend-store sfs \
-              --debug-rgw 1
+              --debug-rgw 1)
 
           sleep 5
           # ^ This is needed to give the gateway some time to boot up.
           # It also serves to make sure the gateway takes no more than 5 seconds
           # to boot up.
 
-          echo "Running Tests:\n"
+          echo "Running Smoke Tests:"
           cd ceph/qa/rgw/store/sfs/tests
-          ./run-tests.sh
+          ./sfs-smoke-test.sh 127.0.0.1:7480
+
+          docker kill $CONTAINER
```

Comment:
Optional: docker run --rm an option?

Reply:
Good point, but that doesn't do the same thing as intended here. `docker run --rm` will remove the container after it has been terminated.
The point of the `docker kill` here is to terminate the container in the first place, so I can start a new one for the next test suite. This avoids running multiple test suites on the same container, potentially leaving a dirty environment for the next test suite, causing failures (e.g. in case of ENOSPC).